sqlparser/parser.go

```diff
@@ -473,18 +473,41 @@ tailloop:
 		case tt == ReservedWordToken && s.Token() == "as":
 			CopyToken(s, &result.Body)
 			NextTokenCopyingWhitespace(s, &result.Body)
-			if firstAs {
-				// Add the `RoutineName` token as a convenience, so that we can refer to the procedure/function name
-				// from inside the procedure (for example, when logging)
-				if result.CreateType == "procedure" {
-					procNameToken := Unparsed{
-						Type:     OtherToken,
-						RawValue: fmt.Sprintf(templateRoutineName, strings.Trim(result.QuotedName.Value, "[]")),
+			/*
+					    TODO: Fix and re-enable
+						This code add RoutineName for convenience.  So:
+
+						create procedure [code@5420c0269aaf].Test as
+						begin
+							select 1
+						end
+						go
+
+						becomes:
+
+						create procedure [code@5420c0269aaf].Test as
+						declare @RoutineName nvarchar(128)
+						set @RoutineName = 'Test'
+						begin
+							select 1
+						end
+						go
+
+						However, for some very strange reason, @@rowcount is 1 with the first version,
```

Comment:
which @@rowcount are you referring to here ?

Reply:
When we do ExecContext, we get back a res that we can call RowsAffected() on it:

```
	res, err := fixture.DB.ExecContext(ctx, patched)
	require.NoError(t, err)
	rowsAffected, err := res.RowsAffected()
```

For some reason, the RoutineName was adding one to the rows affected.

We don't have that issue when we use querysql.ExecContext because there we need to return the number of rows affected manually (instead of calling `RowsAffected()` on `res`).